test/utils/fireUserEvent.ts

```diff
@@ -0,0 +1,26 @@
+import { fireEvent } from '@mui/internal-test-utils/createRenderer';
+
+function touch(target: Element): void {
+  fireEvent.touchStart(target);
+  fireEvent.touchEnd(target);
+}
+
+const mousePress: (...args: Parameters<(typeof fireEvent)['mouseUp']>) => void = (
+  target,
+  options,
+) => {
+  fireEvent.mouseDown(target, options);
+  fireEvent.mouseUp(target, options);
+  fireEvent.click(target, options);
+};
+
+function keyPress(target: Element, options: { key: string; [key: string]: any }): void {
+  fireEvent.keyDown(target, options);
+  fireEvent.keyUp(target, options);
+}
+
+export const fireUserEvent = {
+  touch,
+  mousePress,
+  keyPress,
+};
```

Comment:
> Can't we use https://testing-library.com/docs/user-event/convenience?

That would be the ultimate goal, but after preliminary testing—it's not plug&play, we need more time and effort to migrate tests to using those APIs.

> Otherwise I would rename these to something that is less confusing, like `fireCustomEvent` or something similar.

I have deprecated the exports to improve the clarity.
I don't think renaming it gives much value given that we'd want to remove those imports for the mentioned convenience API anyway eventually. 🙈